### DISCUSSION TITLE
ShipIt: How do merge commits get handled/break the mirror flow?

### DISCUSSION THREAD

**Discussion by itsdouges** [adeira/universe#6762] [Q&A]
Hello! I have a large monorepo that has been acting on merge conflicts for years. Is that a hard blocker from using ShipIt?

> For this reason Shipit requires linear Git history only (it works with reversed ancestry path without merge commits).

**Comment by mrtnzlml** [comment]
Hello! I tried to describe the reasons for the limitation here (seems like you took the quote from there): https://github.com/adeira/shipit/tree/ab75445f833e9f55962de07f3f2876c90d9d1568#linear-history

I am not very confident about the accuracy of the explanation. What I can say for sure is that it has never really been properly tested and nobody is using it on such repositories as far as I know. Would be interesting to have the first test results though. 😎

**Reply by itsdouges** [reply]
Alright no worries will report back.
